src/sentryFilter/index.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { partition } from "../enumerable";
+
+function hasBreadcrumbs(event) {
+  return !!(event && event?.breadcrumbs && event?.breadcrumbs?.length);
+}
+
+export function containsSensitiveInformation(string) {
+  return !!string.match(/code=.+\b/);
+}
+
+export function redactCodeParameter(string) {
+  return string.replace(/(.+)?(code=[A-z,0-9]+)(\b)?/, "$1code=[REDACTED]$3");
+}
+
+export function redactNavigationCrumbs(crumbs) {
+  return crumbs.reduce((acc, crumb) => {
+    if (crumb.from) {
+      Object.assign(crumb, { from: redactCodeParameter(crumb.from) });
+    }
+
+    if (crumb.to) {
+      Object.assign(crumb, { to: redactCodeParameter(crumb.to) });
+    }
+    return [crumb, ...acc];
+  }, []);
+}
+
+export function filterCodeRedirects(event) {
+  if (!hasBreadcrumbs(event)) return event;
+  const [navCrumbs, otherCrumbs] = partition(event?.breadcrumbs);
```

Comment:
I'm not sure why you need `partition` here when a simple `map` would work fine?

```
event.breadcrumbs = event.breadcrumbs.map((crumb) {
  if (crumb.category !== 'navigation') {
    return crumb;
  }

  if (crumb.to) {
    crumb.to = redactCodeParameter(crumb.to);
  }
  
  if (crumb.from) {
    crumb.from = redactCodeParameter(crumb.from);
  }

  return crumb;
});

return event;
```

Seems like it'd be a simpler implementation, but maybe I'm missing something? There's no need to be non-mutative here.

Reply:
Also, your `partition` function takes two parameters, yet you only supply one here, so I don't think it's really partitioning between navCrumbs and non-navCrumbs — but as I mention above, I'd argue you don't actually need the `partition` in the first place.